api/src/main/java/io/strimzi/api/kafka/model/CustomResourceConditions.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import io.fabric8.kubernetes.client.CustomResource;
+import io.strimzi.api.kafka.model.status.Status;
+
+import java.util.function.Predicate;
+
+public class CustomResourceConditions {
+
+    /**
+     * Returns a predicate that determines if a CustomResource is ready. A CustomResource is
+     * ready if the observedGeneration of its status is equal to the generation of its metadata
+     * and any of the conditions of its status has type:"Ready" and status:"True"
+     *
+     * @param <Y> the type of the custom resources Status
+     * @param <T> the type of the custom resource
+     * @return a predicate that checks if a CustomResource is ready
+     */
+    public static <Y extends Status, T extends CustomResource<?, Y>> Predicate<T> isReady() {
```

Comment:
This could be used with _any_ custom resource, even non-Strimzi ones (which might not have `Ready`, and therefore the returned predicate might never return true). To avoid this perhaps we should only expose public methods for Strimzi resources?

Reply:
I also wonder having these as `static` methods on the CR classes (or perhaps their status classes) might make for better discoverability?

Reply:
Having a signature like `static Predicate<Kafka> isReady()` on the `Kafka` class could be easier to update in future if your definition of readiness changes or needed to be specialised per CR. I think it can't go on the `Status` classes as we need access to `metadata` to get at the `generation`.

Reply:
Moved the predicates to public static methods on the CR classes and made CustomResourceConditions package private